src/scripts/main.js

```diff
@@ -1,3 +1,273 @@
+
+  notificationMessage.textContent = description;
+
+  notification.style.top = `${posTop}px`;
+  notification.style.right = `${posRight}px`;
+
+  notification.setAttribute('data-qa', 'notification');
+
+  notification.appendChild(notificationTitle);
+  notification.appendChild(notificationMessage);
+  document.body.appendChild(notification);
+
+  setTimeout(() => {
+    notification.remove();
+  }, 2000);
+};
+
+const filter = document.querySelectorAll('thead tr th');
+const tableBody = document.querySelector('tbody');
+
+for (let i = 0; i < filter.length; i++) {
+  filter[i].addEventListener('click', () => sortRowsByNumber(i));
+}
+
+tableBody.addEventListener('click', (e) => {
+  const active = e.target.closest('tr');
+
+  document
+    .querySelectorAll('tbody tr')
+    .forEach((x) => x.classList.remove('active'));
+  active.classList.add('active');
+});
+
+const form = document.createElement('form');
+
+const namelabel = document.createElement('label');
+
+namelabel.textContent = 'Name: ';
+
+const nameInput = document.createElement('input');
+
+nameInput.type = 'text';
+nameInput.setAttribute('data-qa', 'name');
+nameInput.required = 'true';
+
+const position = document.createElement('label');
+
+position.textContent = 'position: ';
+
+const positionInput = document.createElement('input');
+
+position.type = 'text';
+positionInput.setAttribute('data-qa', 'position');
+positionInput.required = 'true';
+
+const office = document.createElement('label');
+
+office.textContent = 'office: ';
+
+const officeInput = document.createElement('select');
+
+officeInput.setAttribute('data-qa', 'office');
+officeInput.required = 'true';
+
+const optionTokyo = document.createElement('option');
+
+optionTokyo.value = 'Tokyo';
+optionTokyo.textContent = 'Tokyo';
+officeInput.appendChild(optionTokyo);
+
+const optionSingapore = document.createElement('option');
+
+optionSingapore.value = 'Singapore';
+optionSingapore.textContent = 'Singapore';
+officeInput.appendChild(optionSingapore);
+
+const optionLondon = document.createElement('option');
+
+optionLondon.value = 'London';
+optionLondon.textContent = 'London';
+officeInput.appendChild(optionLondon);
+
+const optionNewYork = document.createElement('option');
+
+optionNewYork.value = 'New York';
+optionNewYork.textContent = 'New York';
+officeInput.appendChild(optionNewYork);
+
+const optionEdinburgh = document.createElement('option');
+
+optionEdinburgh.value = 'Edinburgh';
+optionEdinburgh.textContent = 'Edinburgh';
+officeInput.appendChild(optionEdinburgh);
+
+const optionSanFrancisco = document.createElement('option');
+
+optionSanFrancisco.value = 'San Francisco';
+optionSanFrancisco.textContent = 'San Francisco';
+officeInput.appendChild(optionSanFrancisco);
+
+const age = document.createElement('label');
+
+age.textContent = 'age: ';
+
+const ageInput = document.createElement('input');
+
+ageInput.type = 'text';
+ageInput.setAttribute('data-qa', 'age');
+ageInput.required = 'true';
```

Comment:
The 'required' attribute should be a boolean. Use 'ageInput.required = true;' instead of 'ageInput.required = 'true';'.